cdap-docs/admin-manual/source/installation/replication.rst

```diff
@@ -0,0 +1,247 @@
+.. meta::
+    :author: Cask Data, Inc.
+    :copyright: Copyright © 2017 Cask Data, Inc.
+
+.. _installation-replication:
+
+================
+CDAP Replication
+================
+
+This document lists the detailed steps required for setting up CDAP replication, where one
+CDAP cluster (a *master*) is replicated to one or more additional CDAP *slave* clusters.
+
+**Note:** As described below, CDAP must have :ref:`invalid transaction list pruning disabled 
+<installation-replication-disable-invalid-transaction-list-pruning>`, as this cannot be
+used with replication.
+
+**These steps should be reviewed (and the `Cluster Setup`_ completed) prior to starting CDAP.**
+
+.. _installation-replication-cluster-setup:
+
+Cluster Setup
+=============
+
+CDAP replication relies on the cluster administrator setting up replication on `HBase`_,
+`HDFS`_, `Hive`_, and `Kafka`_.
+
+- It is assumed that CDAP is only running on the master cluster.
+- It is assumed that **you have not started CDAP before any of these steps**.
+
+HBase
+-----
+- Install the relevant ``cdap-hbase-compat`` package on all HBase nodes of your cluster in order
+  to use the replication status coprocessors. Note that due to HBase limitations, these
+  coprocessors cannot be used on HBase 0.96 or 0.98. 
+  
+  Available "compat" packages are:
+
+  - ``cdap-hbase-compat-1.0``
+  - ``cdap-hbase-compat-1.0-cdh``
+  - ``cdap-hbase-compat-1.0-cdh5.5.0``
+  - ``cdap-hbase-compat-1.1``
+  - ``cdap-hbase-compat-1.2-cdh5.7.0``
+
+.. highlight:: xml
+
+- Modify ``hbase-site.xml`` on all HBase nodes to enable HBase replication, and to use the
+  CDAP replication status coprocessors::
+
+    <property>
+      <name>hbase.replication</name>
+      <value>true</value>
+    </property>
+    <property>
+      <name>hbase.coprocessor.regionserver.classes</name>
+      <value>co.cask.cdap.data2.replication.LastReplicateTimeObserver</value>
+    </property>
+    <property>
+      <name>hbase.coprocessor.wal.classes</name>
+      <value>co.cask.cdap.data2.replication.LastWriteTimeObserver</value>
+    </property>
+
+.. highlight:: console
+
+- Modify ``hbase-env.sh`` on all HBase nodes to include the HBase coprocessor in the classpath::
+
+    export HBASE_CLASSPATH="$HBASE_CLASSPATH:/opt/cdap/<hbase-compat-version>/coprocessor/*"
```

Comment:
/opt/cdap will be the path if they have installed the cdap packages themselves. On Cloudera Manager and Ambari, it will be different. @dereklwood may know the path in those environments